sdk/core/azure-core-amqp/docs/architecture.md

```diff
@@ -0,0 +1,53 @@
+# Architecture Docs
+
+## Qpid Proton-J Integration
+
+Qpid Proton-J publishes events and messages via its event-driven process called [`Reactor`][Reactor]. `azure-core-amqp`
+hooks into Qpid Proton-J Reactor via `Handlers`.  [`BaseHandler`][BaseHandler] contains all of the events that can be
+listened to.  `Handlers` can be associated with classes implementing the `Extendable` interface via
+`BaseHandler.setHandler(Extendable, Handler)`.  Proton-J `Connection`, `Session`, `Link`, `Sender` and `Receiver` are
```

Comment:
so `Sender` and `Receiver` are not really each endpoint of  `Link` but are two specializations of `Link` type, right?